src/find-changes.ts

```diff
@@ -44,18 +44,30 @@ export async function getBranchPoint(): Promise<string> {
     }
     const eventData: Buffer = await fs.readFile(eventPath)
     const event = JSON.parse(eventData.toString())
-    if (
-      event &&
-      event.pull_request &&
-      event.pull_request.base &&
-      event.pull_request.base.sha
-    ) {
-      core.info(`Found branch point ${event.pull_request.base.sha}`)
-      return event.pull_request.base.sha as string
+    if (event && event.pull_request) {
+      if (
+        !context.fromOriginalBranchPoint &&
+        event &&
+        event.repository &&
+        event.repository.default_branch
+      ) {
+        const upstream = `origin/${event.repository.default_branch}`
```

Comment:
Why when you merge master into your branch, `origin/${event.repository.default_branch}` != `${event.pull_request.base.sha}`?

Reply:
> when you merge master into your branch

I don't know if it is or not, but that's not the case that we're trying to handle.

We are fixing the case where master has changed and *not* been merged in to the PR.

```
    A         B
    * --------*-------  origin/master
    |\       /        \
    | * --- *   PR 1   \
     \                  \
      * ---------------- *    PR 2 to get merged now
                         |
                  refs/pulls/2/merge (CI runs on this)
```

We have two PRs from the same branch point, one has already been merged to master, and the second is getting merged now.

CI runs using a merge of master into the pull request (`refs/pulls/NUMBER/merge`).
By using `event.pull_request.base.sha`, we compare that merge with commit `A` on master (which equals `event.pull_request.base.sha`), so we have a diff that includes PR1 changes, and we "find" changes in whatever package was changed in PR1, in addition to PR2.

So we use `origin/master` as the comparison point, and do a diff between `refs/pulls/NUMBER/merge` and `origin/master` which is *only* the changes in PR2